src/tests/ftest/container/open_close.py

```diff
@@ -25,62 +25,29 @@
 
 import os
 import traceback
-import sys
-import json
 import uuid
-from avocado import Test
+from apricot import TestWithServers
 
-sys.path.append('./util')
-sys.path.append('../util')
-sys.path.append('../../../utils/py')
-sys.path.append('./../../utils/py')
-
-import AgentUtils
+from daos_api import DaosPool, DaosContainer, DaosApiError
 import server_utils
-import write_host_file
-from daos_api import DaosContext, DaosPool, DaosContainer, DaosApiError
 
-class OpenClose(Test):
+class OpenClose(TestWithServers):
     """
     Tests DAOS container open/close function with handle parameter.
+    :avocado: recursive
     """
     def __init__(self, *args, **kwargs):
         super(OpenClose, self).__init__(*args, **kwargs)
         self.container1 = None
         self.container2 = None
 
-    def setUp(self):
-        # these are first since they are referenced in teardown
-        self.pool = None
-        self.hostlist = None
-        self.hostlist = self.params.get("test_servers", '/run/hosts/')
-
-        # get paths from the build_vars generated by build
-        with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                               '../../../../.build_vars.json')) as build_file:
-            build_paths = json.load(build_file)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-        self.server_group = self.params.get("name", '/server_config/',
-                                            'daos_server')
-
-        # setup the DAOS python API
-        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
-
-        self.hostfile = write_host_file.write_host_file(self.hostlist,
-                                                        self.workdir)
-
-        self.agent_sessions = AgentUtils.run_agent(self.basepath, self.hostlist)
-        server_utils.run_server(self.hostfile, self.server_group, self.basepath)
-
     def tearDown(self):
         try:
             if self.pool is not None and self.pool.attached:
                 self.pool.destroy(1)
         finally:
             try:
-                if self.agent_sessions:
-                    AgentUtils.stop_agent(self.hostlist, self.agent_sessions)
-                server_utils.stop_server(hosts=self.hostlist)
+                super(OpenClose, self).tearDown()
             except server_utils.ServerFailed:
```

Comment:
Do this make sense to carry forward?   If we fail to stop the servers - for any reason - the test still passes.  It seems like we may miss a failure at some point.  If the test does something that would for some reason make this expected behavior I think there should at least be a comment here, but ideally only passing on specific known ServerFailed exceptions.  This may be separate from this PR.

Reply:
This is all being removed and the `tearDown()` handling of `TestWithServers`:
https://github.com/daos-stack/daos/blob/609633cf121c3b5f66a678dabc63f0e3f6818e8f/src/tests/ftest/util/apricot/apricot/test.py#L161-L168
 is being used instead, which you can see is not ignoring any `server_utils.stop_server()` exceptions at all.

Reply:
Sorry, I was referring to the `except server_utils.ServerFailed:` line.  The `super(OpenClose, self).tearDown()` could raise a 'server_utils.ServerFailed' and this test case would hide that fact, where other ones do not.